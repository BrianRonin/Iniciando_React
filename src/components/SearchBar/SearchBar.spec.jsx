const { render, screen } = require('@testing-library/react')
import userEvent from '@testing-library/user-event'
import { click } from '@testing-library/user-event/dist/click'
import { keyboard } from '@testing-library/user-event/dist/keyboard'
import { type } from '@testing-library/user-event/dist/type'
import { SearchBar } from '.'

// setup function
function setup(jsx) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  }
}

test('render with a setup function', async () => {
  const { user } = setup(<SearchBar />)
})

const props = {
  type: 'text',
  defaultValue: '',
  onchange: () => 'x',
  placeholder: 'digite aqui',
}

describe('<ShearchBar />', () => {
  it('should call handleChange function on each key', () => {
    const user = userEvent.setup()
    render(<SearchBar {...props} />)

    const input = screen.getByPlaceholderText('digite aqui')
    const onchangeInput = jest.spyOn(input, 'onchange', 'get')
    // input.focus()
    // user.keyboard('abc')
    // console.log(onchangeInput)
    // expect(onchangeInput).toHaveBeenCalledTimes(3)
  })
})
