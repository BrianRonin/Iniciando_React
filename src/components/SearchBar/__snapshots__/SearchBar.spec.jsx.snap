// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShearchBar /> should match snapshot 1`] = `
<div>
  <div>
    <input
      placeholder="pesquise aqui"
      value=""
    />
  </div>
</div>
`;
